src/prefect/workers/base.py

```diff
@@ -972,6 +975,7 @@ async def _submit_run(self, flow_run: "FlowRun") -> None:
 
             self._submitting_flow_run_ids.remove(flow_run.id)
         else:
+            self._submitting_flow_run_ids.remove(flow_run.id)
```

Comment:
After looking at this more closely, since we're removing from `_submitting_flow_run_ids` in both branches, you simplify this by moving the removal outside of the `if/else` statement like this:
```python
if ready_to_submit:
     ...
else:
     self._release_limit_slot(flow_run.id)
self._submitting_flow_run_ids.remove(flow_run.id)
```